diff --git a/gunicorn/util.py b/gunicorn/util.py
index e919d53..a854eb7 100644
--- a/gunicorn/util.py
+++ b/gunicorn/util.py
@@ -14,6 +14,7 @@ except ImportError:
     # Python on Solaris compiled with Sun Studio doesn't have ctypes
     ctypes = None
 
+import errno
 import fcntl
 import os
 import pkg_resources
@@ -146,6 +147,13 @@ def set_owner_process(uid,gid):
             # groups like on osx or fedora
             os.setgid(-ctypes.c_int(-gid).value)
 
+        # Also drop supplemental groups
+        try:
+            os.setgroups([])
+        except OSError, e:
+            if e.errno != errno.EPERM:
+                raise
+
     if uid:
         os.setuid(uid)
 
